cns/restserver/internalapi.go

```diff
@@ -14,3 +19,58 @@ func (service *HTTPRestService) GetPartitionKey() (dncPartitionKey string) {
 	service.RUnlock()
 	return
 }
+
+// This API will be called by CNS RequestController on CRD update.
+func (service *HTTPRestService) CreateOrUpdateNetworkContainerInternal(req cns.CreateNetworkContainerRequest) int {
```

Comment:
Good idea. I didnt want to change existing code which uses int to return the error code. Also since some of the APIs are shared , hence I followed the same pattern. eg. saveNetworkContainerGoalState. With above approach I need to make changes to existing API as well.

Reply:
Yeah, I think this is appropriate then if it follows the existing convention 👍